src/users/users.service.ts

```diff
@@ -0,0 +1,36 @@
+import { HttpException, HttpStatus, Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import * as bcrypt from 'bcrypt';
+
+import { CreateUserInput } from './dto/create-user.input';
+import { User } from './entities/user.entity';
+
+@Injectable()
+export class UsersService {
+  constructor(
+    @InjectRepository(User)
+    private usersRepository: Repository<User>,
+  ) {}
+
+  async create(createUserInput: CreateUserInput): Promise<User> {
+    try {
+      if (!createUserInput.login) {
```

Comment:
You have required field login so this case won't be reached anyway. Also math rand can generate duplicates. So if duplicate were generated there will be an error from db as login key is unique. Its better to use uuid/guid there.